[REP-2004] add to the motivation to address common questions

wjwwood opened the pull request:
@maryaB-osr can you please take a look? After we iterate on it, I can ask for feedback more broadly.

maryaB-osr commented on rep-2004.rst:
```diff
@@ -13,7 +13,7 @@ Abstract
 This REP describes a set of categories meant to convey the quality or maturity of packages in the ROS ecosystem.
 Inclusion in a category, or quality level, is based on the policies to which a package adheres.
-The categories address version, change control, documentation, testing, dependency and platform support policies.
+The categories address policies about things including version, change control, documentation, testing, dependencies, platform support and others.
```
```suggestion
The categories address policies about versioning, change control, documentation, testing, dependencies, platform support and security.
```

maryaB-osr commented:
To me, it seems like the Motivation and Antigoal sections are switching back and forth between each other, purpose-wise.

I'd rewrite both sections to stay as focused on their purpose as possible, and remove some repeating points between the two:

```
Motivation
==========

The purpose of this REP is to provide standard guidelines for evaluating package quality, for both maintainers and consumers.

Package maintainers can use these guidelines to establish and document their own specific policies addressing how their packages achieve certain quality levels.

Consumers can use the guidelines in the REP, as well as the corresponding policies defined by maintainers, to set expectations on package quality and better evaluate whether a package or its dependencies meet the standards of use in their projects.

By setting these goals and expectations for maintainers and consumers, respectively, the guidelines will also encourage better quality across the ROS ecosystem.

Antigoals
^^^^^^^^^

The motivation behind this REP does not include:

* Enforcing quality levels on maintainers

    * No maintainer is required to abide by any of the guidelines in this REP

* Dictating policy implementation specifics necessary to achieve a quality level

    * Maintainers can come up with their own policies or follow an existing set of guidelines for achieving these quality levels [see "Reference Implementation"] (add anchor before reference implementation section)
    #### Not sure why they’re looking at the reference implementation here though? That section isn't where the option to justify the lack of meeting one of the policies is explained. Wherever that is, is where this should link to 
    * Policy requirements (which tools to use, specific thresholds, etc.) are *intentionally generic*

* Enforcing the community to approve of policies documented by package maintainers

    * Policies that are skipped or altered with justifications require evaluation by the consumer to see if that reasoning is acceptable for their standards of use
    * Enforcement cannot, therefore, be done in an automated fashion in all cases
    * Instead, peer reviewed lists of packages, where maintainers can take their [quality declaration](link to Reference Implementation anchor again) and have their peers verify that their claims/justifications are truthful/reasonable, respectively, can be created 

The outcome of this REP should be that maintainers have looked at each guideline, thought about how their practices address each, adjusted their policies if needed, and then documented their policies and justifications.
The goal being to improve documentation of these policies and improve communication of these quality characteristics between maintainers to consumers.
```

wjwwood commented:
Thanks @maryaB-osr I updated the pr to reflect your suggested wording, it looks clear to me. I fixed up the links, I think they should work based on https://docutils.sourceforge.io/docs/user/rst/quickref.html#implicit-hyperlink-targets. Can you have another brief look?

After which I'll take this out of draft mode and ask for feedback.

maryaB-osr commented on rep-2004.rst:
```diff
@@ -13,18 +13,41 @@ Abstract
+
+    * Policies that are skipped or altered with justifications require evaluation by the consumer to see if that reasoning is acceptable for their standards of use
+    * Enforcement cannot, therefore, be done in an automated fashion in all cases
+    * Instead, peer reviewed lists of packages, where maintainers can take their `quality declaration <Reference Implementation>`_ and have their peers verify that their claims/justifications are truthful/reasonable, respectively, can be created 
```
I generated the page and it works like this:
```suggestion
    * Instead, peer reviewed lists of packages, where maintainers can take their `quality declaration <#reference-implementation>`_ and have their peers verify that their claims/justifications are truthful/reasonable, respectively, can be created 
```

maryaB-osr commented on rep-2004.rst:
```diff
@@ -13,18 +13,41 @@ Abstract
+    * Enforcement cannot, therefore, be done in an automated fashion in all cases
+    * Instead, peer reviewed lists of packages, where maintainers can take their `quality declaration <Reference Implementation>`_ and have their peers verify that their claims/justifications are truthful/reasonable, respectively, can be created 
+
+The outcome of this REP should be that maintainers have looked at each guideline, thought about how their practices address each, adjusted their policies if needed, and then documented their policies and justifications.
+The goal being to improve documentation of these policies and improve communication of these quality characteristics between maintainers to consumers.
```
```suggestion
```

maryaB-osr commented on rep-2004.rst:
```diff
@@ -13,18 +13,41 @@ Abstract
-By providing these rough goals for package maintainers to strive towards, the categories will also encourage better quality across the ROS ecosystem.
+Consumers can use the guidelines in the REP, as well as the corresponding policies defined by maintainers, to set expectations on package quality and better evaluate whether a package or its dependencies meet the standards of use in their projects.
+
+By setting these goals and expectations for maintainers and consumers, respectively, the guidelines will also encourage better quality across the ROS ecosystem.
```
I think the last two sentences from the antigoal section belong here (feel free to ignore this one if you think they're better off below, I just think since they go back to talking about goals they shouldn't be in a section with an opposite purpose)

```suggestion
The outcome of this REP should be that maintainers who want to evaluate their packages' quality have looked at each guideline, thought about how their practices address each, adjusted their policies if needed, and then documented their policies and justifications. 

In turn, the documentation of policies and communication of quality characteristics between maintainers and consumers will improve.
By setting these goals and expectations for maintainers and consumers, respectively, the guidelines will also encourage better quality across the ROS ecosystem.
```

maryaB-osr commented on rep-2004.rst:
```diff
@@ -13,18 +13,41 @@ Abstract
+
+* Enforcing quality levels on maintainers
+
+    * No maintainer is required to abide by any of the guidelines in this REP
```
```suggestion
  * No maintainer is required to abide by any of the guidelines in this REP
```

maryaB-osr commented on rep-2004.rst:
```diff
@@ -13,18 +13,41 @@ Abstract
+* Dictating policy implementation specifics necessary to achieve a quality level
+
+    * Maintainers can come up with their own policies or follow an existing set of guidelines for achieving these quality levels, see `Reference Implementation`_ for examples of existing guidelines
+    * Policy requirements (which tools to use, specific thresholds, etc.) are *intentionally generic*
```
```suggestion
  * Policy requirements (which tools to use, specific thresholds, etc.) are *intentionally generic*
```

maryaB-osr commented on rep-2004.rst:
```diff
@@ -13,18 +13,41 @@ Abstract
+
+* Enforcing the community to approve of policies documented by package maintainers
+
+    * Policies that are skipped or altered with justifications require evaluation by the consumer to see if that reasoning is acceptable for their standards of use
+    * Enforcement cannot, therefore, be done in an automated fashion in all cases
```
```suggestion
  * Policies that are skipped or altered with justifications require evaluation by the consumer to see if that reasoning is acceptable for their standards of use
  * Enforcement cannot, therefore, be done in an automated fashion in all cases
```

ros-discourse commented:
This pull request has been mentioned on **ROS Discourse**. There might be relevant details there:

https://discourse.ros.org/t/rfc-rep-2004-package-quality-categories/13150/3

wjwwood commented:
I'll merge this after waiting a few days to see if there are any comments.

gbiggs commented on rep-2004.rst:
```diff
@@ -13,18 +13,42 @@ Abstract
+
+Consumers can use the guidelines in the REP, as well as the corresponding policies defined by maintainers, to set expectations on package quality and better evaluate whether a package or its dependencies meet the standards of use in their projects.
+
+The outcome of this REP should be that maintainers who want to evaluate their packages' quality have looked at each guideline, thought about how their practices address each, adjusted their policies if needed, and then documented their policies and justifications. 
```
I'm not clear on the definitions of "policy" and "practice" in this sentence. Does this work better?

```suggestion
The outcome of this REP should be that maintainers who want to evaluate their packages' quality have looked at each guideline, thought about how their policies address each, adjusted their policies if needed, and then documented their policies and justifications. 
```

wjwwood commented:
Ok, thanks everyone.
